### PR TITLE
Add checklists.md to the documentation

### LINKED ISSUE #2319: Add a release checklist to document the release process
_Idea from the community call._

### PR DESCRIPTION
**Pull request by odow**
Closes #2319

Once we have https://github.com/JuliaDocs/Documenter.jl/issues/2314, we can add an additional step to run the regression tests.

### CODE REVIEW

File: docs/src/checklists.md
Comment (mortenpi) [inline docs/src/checklists.md:30]: I don't think we have any links to JuliaHub?

File: docs/src/checklists.md
Comment (mortenpi) [inline docs/src/checklists.md:24]: I've recently started using JuliaHub's registration UI for tagging. We could document both ways?

File: docs/src/checklists.md
Comment (mortenpi) [inline docs/src/checklists.md:19]:
```suggestion
 - [ ] Update `CHANGELOG.md`, following the existing style (in particular, make sure that the change log for this version has the correct version number and date)
 - [ ] Run `make changelog`, to make sure that all the issue references in `CHANGELOG.md` are up to date.
```

**Review by mortenpi — COMMENTED** [review]
This LGTM! I wonder if @fredrikekre can think of something we'd might want to add?

File: docs/src/checklists.md
Comment (odow) [reply docs/src/checklists.md:24]: I don't know how that works

File: docs/src/checklists.md
Comment (odow) [reply docs/src/checklists.md:31]: Ah. I just copied this from JuMP

File: docs/src/checklists.md
Comment (mortenpi) [inline docs/src/checklists.md:33]: I hope I got the indentation right here:

```suggestion
 - [ ] After merging this pull request, tag the release. There are two options for this:
 
       1. [Comment `[at]JuliaRegistrator register` on the GitHub commit.](https://github.com/JuliaRegistries/Registrator.jl#via-the-github-app)
       2. Use [JuliaHub's package registration feature](https://help.juliahub.com/juliahub/stable/contribute/#registrator) to trigger the registration.
 
       Either of those should automatically publish a new version to the Julia registry.
       Once registered, the `TagBot.yml` workflow should create a tag, and rebuild the
       documentation for this tag. These steps can take quite a bit of time (1
       hour or more), so don't be surprised if the new documentation takes a
       while to appear.
```

**Comment by odow** [comment]
Tweaked the indentation by trial-and-error

**Review by mortenpi — APPROVED** [review]
Let's merge this?

**Comment by odow** [comment]
Good by me. Once it's there it's a lot easier to make changes if we find more steps are needed.

**Review by fredrikekre — APPROVED** [review]

File: docs/src/checklists.md
Comment (fredrikekre) [inline docs/src/checklists.md:30]: Perhaps add TagBot release notes here? A link to the changelog at least.

Comment (mortenpi) [reply docs/src/checklists.md:30]: I added that into `TagBot.yml`: https://github.com/JuliaDocs/Documenter.jl/blob/efb3ab4d176023b50fe5d6c49a3d87096d67760c/.github/workflows/TagBot.yml#L19-L28 and it seems to generate correct release notes: https://github.com/JuliaDocs/Documenter.jl/releases

Comment (fredrikekre) [reply docs/src/checklists.md:30]: Ah cool, didn't know that was possible!

File: docs/src/checklists.md
Comment (goerz) [inline docs/src/checklists.md:28]: Again, what is "this pull request"? If it's the PR for the release branch, I would usually tag the release *before* merging the PR. This is in case there is any kind of problem like a broken compat bound that might only show up in the registration itself. If there is a problem, I would amend the release branch, force push it, and retrigger the registration.

Once the registration goes through and Tagbot tags the commit, I would then merge the PR (using a fast-forward merge)

The one annoyance I have with merging the release-PR *after* the release is tagged is that it then shows up as a "closed PR" for the *next* release in the release notes that Github auto-generates.

File: docs/src/checklists.md
Comment (mortenpi) [reply docs/src/checklists.md:14]: We have only really used release branches when we had breaking(-ish) changes on master, and we wanted to tag some changes without a 0.x minor version bump, just a patch bump. Since we don't need to worry about that anymore, I don't think we'll be using release branches (unless some need comes up where we have to "LTS" some branch).

Comment (goerz) [reply docs/src/checklists.md:14]: Ok, makes sense… Case 1 is what I meant by "release branch" (not a long-running branch for backports or anything like that.

File: docs/src/checklists.md
Comment (goerz) [inline docs/src/checklists.md:10]:
```suggestion
In preparation for a release, use the following checklist. These steps should be performed on a branch with an open pull request, either for a topic branch, or for a new branch `release-1.y.z` ("Release version 1.y.z") if multiple changes have accumulated on the `master` branch since the last release.
```

File: docs/src/checklists.md
Comment (goerz) [inline docs/src/checklists.md:28]:
```suggestion
 - [ ] After merging the pull request, tag the release. There are two options for this:
```

I suppose tagging *after* the pull request is merged is a smoother workflow when everything goes well, but there's still the small chance that the release will fail for some reason. If we're okay with amending the release commit (which is now on `master`) in that case, force-pushing it, and retriggering the release, it wouldn't be a problem. Failures should be very rare, especially if we make sure that we have something in the CI that checks that `compat` bounds are valid (or whatever other requirements the Registrator has for auto-merging an update), so maybe this isn't really anything to worry about

**Pull request merged**
